.github/workflows/ci.yml

```diff
@@ -0,0 +1,149 @@
+      run: ./checker/bin-devel/test-${{ matrix.script }}.sh
+
+  # The remaining tests for JDK 21. Separate from `sanity` to allow parallelism with `otheros`.
+  remainder:
+    name: Test ${{ matrix.script }} on JDK ${{ matrix.java_version }}
+    runs-on: ubuntu-latest
+    needs: sanity
+    permissions:
+      contents: read
+    strategy:
+      matrix:
+        script: ['typecheck-part1', 'typecheck-part2', 'guava', 'plume-lib', 'jspecify-conformance']
+        java_version: [21]
+    env:
+      JAVA_VERSION: ${{ matrix.java_version }}
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up JDK ${{ matrix.java_version }}
+      uses: actions/setup-java@v4
+      with:
+        java-version: ${{ matrix.java_version }}
+        distribution: 'temurin'
+
+    # Configure Gradle for optimal use in GitHub Actions, including caching of downloaded dependencies.
+    # See: https://github.com/gradle/actions/blob/main/setup-gradle/README.md
+    - name: Setup Gradle
+      uses: gradle/actions/setup-gradle@v3.5.0
+
+    - name: Run test script checker/bin-devel/test-${{ matrix.script }}
+      run: ./checker/bin-devel/test-${{ matrix.script }}.sh
+
+  otherjdks:
+    name: Test ${{ matrix.script }} on JDK ${{ matrix.java.version }}
+    runs-on: ubuntu-latest
+    needs: remainder
+    permissions:
+      contents: read
+    strategy:
+      matrix:
+        # jspecify-conformance only tested on JDK 21
+        script: ['cftests-junit', 'cftests-nonjunit', 'typecheck-part1', 'typecheck-part2', 'guava', 'plume-lib']
+        # JDK 21 used by sanity before
+        java: [{version: '8', experimental: false},
+          {version: '11', experimental: false},
+          {version: '17', experimental: false},
+          {version: '22', experimental: true},
+          {version: '23-ea', experimental: true},
+          {version: '24-ea', experimental: true}]
+    env:
+      JAVA_VERSION: ${{ matrix.java.version }}
+    continue-on-error: ${{ matrix.java.experimental }}
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up JDK ${{ matrix.java.version }}
+      uses: actions/setup-java@v4
+      with:
+        java-version: ${{ matrix.java.version }}
+        distribution: 'temurin'
+
+    # Configure Gradle for optimal use in GitHub Actions, including caching of downloaded dependencies.
+    # See: https://github.com/gradle/actions/blob/main/setup-gradle/README.md
+    - name: Setup Gradle
+      uses: gradle/actions/setup-gradle@v3.5.0
+
+    - name: Run test script checker/bin-devel/test-${{ matrix.script }}
+      run: ./checker/bin-devel/test-${{ matrix.script }}.sh
+
+  # Sanity tests on Windows and MacOS.
+  otheros:
+    name: Test ${{ matrix.script }} on JDK ${{ matrix.java_version }} on ${{ matrix.os }}
+    runs-on: "${{ matrix.os }}"
+    needs: sanity
+    permissions:
+      contents: read
+    strategy:
+      matrix:
+        os: ['windows-latest', 'macos-latest']
```

Comment:
I remember for mac os, we need to install timeout by `brew install coreutils`. That's why current CI are failing.

Reply:
Thanks! I think this is working now.